streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -62,11 +70,14 @@ private V computeNewValue(K key, V value) {
         return newValue;
     }
 
-    private class KTableFilterProcessor extends AbstractProcessor<K, V> {
+    private class KTableFilterProcessor extends AbstractProcessor<K, Change<V>> {
 
         @Override
-        public void process(K key, V value) {
-            context().forward(key, computeNewValue(key, value));
+        public void process(K key, Change<V> change) {
+            V newValue = computeValue(key, change.newValue);
+            V oldValue = sendOldValues ? computeValue(key, change.oldValue) : null;
+
+            context().forward(key, new Change<>(newValue, oldValue));
```

Comment:
This may be another place we can optimize away temporary Change object creation with Struct.